mmdet/models/losses/focal_loss.py

```diff
@@ -179,3 +179,67 @@ def forward(self,
         else:
             raise NotImplementedError
         return loss_cls
+
+
+@LOSSES.register_module()
+class BinaryFocalLoss(nn.Module):
+
+    def __init__(self,
+                 alpha: float = 0.25,
+                 beta: float = 4,
+                 gamma: float = 2,
+                 pos_weight: float = 0.5,
+                 neg_weight: float = 0.5,
+                 sigmoid_clamp: float = 1e-4,
+                 ignore_high_fp: float = -1.):
+
+        """`Focal Loss <https://arxiv.org/abs/1708.02002>`_
+        """
+        super(BinaryFocalLoss, self).__init__()
+        self.alpha = alpha
+        self.beta = beta
+        self.gamma = gamma
+        self.pos_weight = pos_weight
+        self.neg_weight = neg_weight
+        self.sigmoid_clamp = sigmoid_clamp
+        self.ignore_high_fp = ignore_high_fp
+
+    def forward(self, inputs, targets, pos_inds, avg_factor=None):
+
+        """
+        Args:
+            inputs:  (sum_l N*Hl*Wl,)
```

Comment:
add type and other details

Reply:
Hey, thanks for the comment. Currently there still has a ton of work to do with this implementation. However, I am in the competition and the due is around the corner, so I made the PR before further polishing. Most importantly, I haven't had a decent result yet (it is still training on a single GPU). So I think I will address these comments once I have some meaningful results. Thanks again.